player-vs-computer.js

```diff
@@ -0,0 +1,240 @@
+      console.log(
+        `\n${input} is not a valid number. Enter something else please!\n`
+      );
+      await getBottomRange();
+    } else if (isNumeric(input) === true && parseFloat(input) % 1 != 0) {
+      console.log("\nIntegers only, please!\n");
+      await getBottomRange();
+    } else if (isNumeric(input) === true && parseFloat(input) < 0) {
+      console.log("\nNo negative numbers, please. Let's try again.\n");
+      await getBottomRange();
+    } else {
+      return;
+    };
+  };
+
+  await getTopRange();
+  //get range top
+  async function getTopRange() {
+    let rangeTopInput = await ask("\nWhat is the top number of the range?\n\n");
+    rangeTop = parseInt(rangeTopInput);
+    await checkTopRange(rangeTopInput);
+    return;
+  };
+
+  // ensure range top is valid
+  async function checkTopRange(input) {
+    if (!isNumeric(input)) {
+      console.log(
+        `\n${input} is not a valid number, ${name}. Enter something else please!\n`
+      );
+      await getTopRange();
+    } else if (isNumeric(input) === true && parseFloat(input) % 1 != 0) {
+      console.log("\nIntegers only, please!\n");
+      await getTopRange();
+    } else if (isNumeric(input) === true && parseInt(input) < 0) {
+      console.log("\nNo negative numbers, please. Let's try again.\n");
+      await getTopRange();
+    } else if (parseInt(input) < rangeBottom) {
+      console.log(
+        "\nThe top end of the range has to be higher than the low end. We both know you know that.\n"
+      );
+      await getTopRange();
+    } else {
+      return;
+    };
+  };
+
+  // function to prevent user from entering anything other than a "numerical string", e.g. "dghdfgd"
+  function isNumeric(num) {
+    return !isNaN(parseFloat(num)) && isFinite(num);
+  };
+
+  //generate computer's 'secret number'
+  let computerNum = randomNum(rangeBottom, rangeTop);
+
+  //generate computer's guess
+  let computerGuess = binarySearch(rangeBottom, rangeTop);
+  console.log("\nOk, I have selected my secret number out of that range.\n");
+
+  await getPlayerNum();
+
+  async function getPlayerNum() {
+    playerNum = await ask(
+      "Now, what is your secret number?\nI won't peek, I promise...\n\n"
+    );
+    await checkPlayerNum(playerNum);
+  };
+
+  async function checkPlayerNum(input) {
+    if (!isNumeric(input)) {
+      console.log(
+        `${input} is not a valid number, ${name}. Enter something else please!\n\n`
+      );
+      await getPlayerNum();
+    } else if (isNumeric(input) === true && parseFloat(input) % 1 != 0) {
+      console.log("\nIntegers only, please!\n");
+      await getPlayerNum();
+    } else if (parseInt(input) < rangeBottom || parseInt(input) > rangeTop) {
+      console.log(
+        "\nThe command line wizard told me your number is outside of the range we agreed upon. That is deeply unchill of you. Please pick a new number within the range.\n"
+      );
+      await getPlayerNum();
+    };
+  };
```

Comment:
This function also is one that could be looked at for refactoring and combining with the other two. A lot of these checks are being done the same way, so creating one function to do all of the similar ones could neaten your code up